### PR TITLE
Feature/mobx - adds mobx support to @tldraw/core

### PR DESCRIPTION
**Pull request by noah79**
This PR adds observer support for all internal components in @tldraw/core along with autorun support for the layout effect in  useCameraCss.  These changes should have no impact on existing consumers of the library, but if <Renderer> is passed observable page/pageState data, it will now automatically update if any observed values change.

### CODE REVIEW

**Comment by steveruizok** [comment]
Text tool causes an error, investigating.

File: examples/core-example/src/stores.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        zoom: 1,
+      },
+    } = opts
+    this.id = id
+    this.camera = camera
+    this.selectedIds = selectedIds
+    makeAutoObservable(this)
+  }
+
+  @action setHoveredId = (id: string | undefined) => {
```
Comment (cagataycali) [inline examples/core-example/src/stores.ts:52]: Why we expect this id is could be undefined here?

File: examples/core-example/src/stores.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    this.hoveredId = id
+  }
+
+  @action setSelectedIds = (id: string) => {
+    if (!this.selectedIds.includes(id)) {
```
Comment (cagataycali) [inline examples/core-example/src/stores.ts:57]: What if we had already **selectedIds** but not included given id

**Comment by steveruizok** [comment]
Includes a few other cleanup changes and chores on core too, including removing unused utils and splitting the spline-related utils into a separate package, in the hopes of slimming down core to balance the addition of mobx.

**Pull request merged**